backend/src/auth/auth.controller.ts

```diff
@@ -1,22 +1,25 @@
-import { Controller, Get, UseGuards, Req, Res } from '@nestjs/common';
+import { Body, Controller, Get, UseGuards, Req, Res } from '@nestjs/common';
 import { ApiOkResponse, ApiUnauthorizedResponse } from '@nestjs/swagger';
-import { FourtyTwoAuthGuard } from './auth.guard';
+import { FourtyTwoAuthGuard } from './guard/FourtyTwo.guard';
+import { AuthService } from './auth.service';
+import { AuthDto } from './dto';
 
 @Controller('auth')
 export class AuthController {
-  @Get('fourtytwo/login')
-  @UseGuards(FourtyTwoAuthGuard)
-  @ApiOkResponse({ description: 'Try logging in using 42 oauth' })
-  @ApiUnauthorizedResponse({ description: 'Login failed.' })
-  handle42Oauth() {}
+	constructor(private authService: AuthService) {}
 
-  @Get('fourtytwo/callback')
-  @UseGuards(FourtyTwoAuthGuard)
-  @ApiOkResponse({ description: '42 oauth callback url' })
-  @ApiUnauthorizedResponse({ description: 'Login failed.' })
-  async handle42Login(@Res() res: any, @Req() req: any): Promise<string> {
-    console.log('req.user', req.user);
-    res.cookie('access_token', req.user).redirect('/hello');
-    return 'OK!';
-  }
+	@Get('fourtytwo/login')
+	@UseGuards(FourtyTwoAuthGuard)
+	@ApiOkResponse({ description: 'Try logging in using 42 oauth' })
+	@ApiUnauthorizedResponse({ description: 'Login failed.' })
+	handle42Oauth() {}
+
+	@Get('fourtytwo/callback')
+	@UseGuards(FourtyTwoAuthGuard)
+	@ApiOkResponse({ description: '42 oauth callback url' })
+	@ApiUnauthorizedResponse({ description: 'Login failed.' })
+	async handle42Login(@Res() res: any, @Req() req: any): Promise<string> {
+		this.authService.handle42Login(res, req.user);
+		return 'OK';
```

Comment:
Might not need this return. 😃  Since we call "authService.handle42Login" and do the redirection there.